From 129e49234fb0a866fd63be9f48c27b8c2884e902 Mon Sep 17 00:00:00 2001
From: Wang ShaoBo <bobo.shaobowang@huawei.com>
Date: Fri, 31 Dec 2021 15:59:50 +0800
Subject: [PATCH] kabi: reserve space for fwnode.h
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 129e49234fb0a866fd63be9f48c27b8c2884e902


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4KFY7?from=project-issue
CVE: NA

-------------------------------

reserve space for struct fwnode_handle and fwnode_operations.

Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/fwnode.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/linux/fwnode.h b/include/linux/fwnode.h
index 9506f8ec0974..06eeb7266401 100644
--- a/include/linux/fwnode.h
+++ b/include/linux/fwnode.h
@@ -10,6 +10,7 @@
 #define _LINUX_FWNODE_H_
 
 #include <linux/types.h>
+#include <linux/kabi.h>
 
 struct fwnode_operations;
 struct device;
@@ -18,6 +19,8 @@ struct fwnode_handle {
 	struct fwnode_handle *secondary;
 	const struct fwnode_operations *ops;
 	struct device *dev;
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
 };
 
 /**
@@ -147,6 +150,10 @@ struct fwnode_operations {
 				    struct fwnode_endpoint *endpoint);
 	int (*add_links)(const struct fwnode_handle *fwnode,
 			 struct device *dev);
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+	KABI_RESERVE(3)
+	KABI_RESERVE(4)
 };
 
 #define fwnode_has_op(fwnode, op)				\
-- 
2.26.2

